### PR TITLE
Improve melee code, fix some melee issues

### LINKED ISSUE #6089: Punch Sound Missing
Not sure if this is broke or just a missing feature, but when you punch another player there's no punch sound only a sound if you miss.

### LINKED ISSUE #6423: You are unable to melee attack nor pet mobs.
### Environment
- Game Version: 21042400
- OS Version: Windows 10

### Detailed description of bug
Clicking on mobs doesn't make you melee attack them nor pet (where it should be applicable) them. This means that players are unable to attack mobs either for self-defense, butchering, and general harming and killing. Possibly related to #6367.

### Steps to reproduce
1. Find a corgi, cat, any domesticated animal and while adjacent to it, click on it with an empty hand on help intent.
2. Watch as you don't pet it.
3. Switch to harm intent and click on the same animal while still adjacent to it with or without an empty hand.
4. Watch as you don't attack it.

### PR DESCRIPTION
**Pull request by NoooneyDude**
- Improves melee code.
- Fixes melee bugs.
    - Fixes being unable to melee mobs with the old `LivingHealthBehaviour` component. Fixes #6423.
    - You can now punch windows and grilles (and walls, which may not be desirable but is harmless enough).
- Re-adds punch sounds. Fixes #6089.

### CODE REVIEW

**Comment by NoooneyDude** [comment]
As for the Codacy complaint: I don't feel like namespacing `Integrity` in this MR: I'll have to update 60 scripts - though it is good that Codacy cares.

File: UnityProject/Assets/Scripts/Items/Weapons/Meleeable.cs [hunk 1]
```diff
@@ -1,144 +1,92 @@
...
+	/// Allows an object or tiles to be attacked by melee.
 	/// </summary>
-	private static readonly HashSet<LayerType> attackableLayers = new HashSet<LayerType>(
-		new[]
-		{
+	public class Meleeable : MonoBehaviour, IPredictedCheckedInteractable<PositionalHandApply>
+	{
+		[SerializeField]
+		private bool isMeleeable = true;
+		// If it has this component, isn't it assumed to be meleeable? Is this still true for tilemaps?
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Items/Weapons/Meleeable.cs:15]: yeah what is this field?

Comment (NoooneyDude) [reply UnityProject/Assets/Scripts/Items/Weapons/Meleeable.cs:15]: No idea. It is only read in one place, though I don't know how many objects are currently using it (or tiles).

**Comment by corp-0** [comment]
Rider should be able to get usages of the serialized field. This is weird, like I thought the jobType enum was bad but this is another level of bad lol

File: UnityProject/Assets/Scripts/Items/Weapons/Meleeable.cs (hunk 1 above)
Comment (corp-0) [reply UnityProject/Assets/Scripts/Items/Weapons/Meleeable.cs:15]: Rider should be able to get usages of the serialized field. This is weird, like I thought the jobType enum was bad but this is another level of bad lol

Comment (AutumnNova) [reply UnityProject/Assets/Scripts/Items/Weapons/Meleeable.cs:15]: ah this, it is used to prevent melee attacks on items and objects

Comment (NoooneyDude) [reply UnityProject/Assets/Scripts/Items/Weapons/Meleeable.cs:15]: I've removed the bad dependency of `Integrity` requiring the `Meleeable` component. Instead, new / modified objects should just not attach the `Meleeable` component if it isn't meleeable as opposed to setting the `isMeleeable` property on the component. Though the field is still available for compatibility with current objects.

**Comment by corp-0** [comment]
Looks good and as discussed on discord, seems like all changes are what is intended. Merging

**Comment by PerfectTangent** [comment]
this may have caused some unintended bugs with mining. PKAs and pickaxes do not work properly

**Comment by PerfectTangent** [comment]
Peek_15-05-2021_00-36

**Pull request merged**